modules/cudaarithm/src/cuda/polar_cart.cu

```diff
@@ -192,6 +276,49 @@ namespace
         ymat(y, x) = mag_val * sin_a;
     }
 
+    template <typename T, bool useMag>
+    __global__ void polarToCartDstInterleavedImpl_(const GlobPtr<T> mag, const GlobPtr<T> angle, GlobPtr<typename MakeVec<T, 2>::type > xymat, const T scale, const int rows, const int cols)
```

Comment:
If you use `PtrStep<T>` for `mag`, `angle` and `xymat` then you pass them directly to this function instead of using the intermediate `GpuMat_<T>` with reshape.  Additionaly if `angle` is a `PtrStepSz<T>` then you don't need to pass `cols` and `rows` seperately. e.g.

```
    __global__ void polarToCartDstInterleavedImpl_(const PtrStep<T> mag, const PtrStepSz<T> angle, PtrStep<typename MakeVec<T, 2>::type > xymat, const T scale, const int rows, const int cols)
    {
        typedef typename MakeVec<T, 2>::type T2;
        const int x = blockDim.x * blockIdx.x + threadIdx.x;
        const int y = blockDim.y * blockIdx.y + threadIdx.y;

        if (x >= angle.cols || y >= angle.rows)
            return;
```

You can also make this adjustment to all the other `polarToCart` kernel calls including the existing one.